src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -1553,59 +1573,71 @@ export class WorkspacesUsecases {
         updatedAt: new Date(),
       });
 
-      await this.workspaceRepository.addUserToWorkspace(workspaceUser);
+      await this.workspaceRepository.addUserToWorkspace(workspaceUser, {
+        transaction,
+      });
 
       await this.teamRepository.addUserToTeam(
         workspace.defaultTeamId,
         user.uuid,
+        { transaction },
       );
 
-      await this.workspaceRepository.createItem({
-        itemId: rootFolder.uuid,
-        workspaceId: workspace.id,
-        itemType: WorkspaceItemType.Folder,
-        context: WorkspaceItemContext.Drive,
-        createdBy: user.uuid,
-        createdAt: new Date(),
-        updatedAt: new Date(),
-      });
+      await this.workspaceRepository.createItem(
+        {
+          itemId: rootFolder.uuid,
+          workspaceId: workspace.id,
+          itemType: WorkspaceItemType.Folder,
+          context: WorkspaceItemContext.Drive,
+          createdBy: user.uuid,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        },
+        { transaction },
+      );
 
-      await this.workspaceRepository.deleteInviteBy({ id: invite.id });
+      await this.workspaceRepository.deleteInviteBy(
+        { id: invite.id },
+        { transaction },
+      );
+
+      await this.adjustOwnerStorage(workspace.id, invite.spaceLimit, 'DEDUCT', {
+        transaction,
+      });
 
-      await this.adjustOwnerStorage(workspace.id, invite.spaceLimit, 'DEDUCT');
+      await transaction.commit();
 
       return workspaceUser.toJSON();
     } catch (error) {
-      let finalMessage = 'There was a problem accepting this invite! ';
-      const rollbackError = await this.rollbackUserAddedToWorkspace(
-        user.uuid,
-        workspace,
-      );
-
-      finalMessage += rollbackError
-        ? rollbackError.message
-        : 'rollback applied successfully';
+      await transaction.rollback();
+      const finalMessage = 'There was a problem accepting this invite!';
 
       Logger.error(
         `[WORKSPACE/ACEPT_INVITE]: Error while accepting user invitation! ${
           (error as Error).message
         }`,
       );
 
+      if (error instanceof BadRequestException) {
+        throw new BadRequestException(finalMessage);
```

Comment:
Why are not we returning the real error message in case of a Bad Request?

Reply:
Since I moved most of the logic inside a try catch to be able to rollback the transaction our BadRequestExceptions are would be catched and then thrown as 500 Internal server Error, so that's why im rethrowing them as BadRequestException